examples/model_checking.py

```diff
@@ -35,115 +36,205 @@ def __init__(self, variables, init, trans):
 # EOC TransitionSystem
 
 
-
-def get_subs(system, i):
-    """Builds a map from x to x@i and from x' to x@(i+1), for all x in system."""
-    subs_i = {}
-    for v in system.variables:
-        subs_i[v] = at_time(v, i)
-        subs_i[next_var(v)] = at_time(v, i+1)
-    return subs_i
-
-
-def get_unrolling(system, k):
-    """Unrolling of the transition relation from 0 to k:
-
-    E.g. T(0,1) & T(1,2) & ... & T(k-1,k)
-    """
-    res = []
-    for i in xrange(k+1):
-        subs_i = get_subs(system, i)
-        res.append(system.trans.substitute(subs_i))
-    return And(res)
-
-
-def get_simple_path(system, k):
-    """Simple path constraint for k-induction:
-    each time encodes a different state
-    """
-    res = []
-    for i in xrange(k+1):
-        subs_i = get_subs(system, i)
-        for j in xrange(i+1, k+1):
-            state = []
-            subs_j = get_subs(system, j)
-            for v in system.variables:
-                v_i = v.substitute(subs_i)
-                v_j = v.substitute(subs_j)
-                state.append(Not(EqualsOrIff(v_i, v_j)))
-            res.append(Or(state))
-    return And(res)
-
-
-def get_k_hypothesis(system, prop, k):
-    """Hypothesis for k-induction: each state up to k-1 fulfills the property"""
-    res = []
-    for i in xrange(k):
-        subs_i = get_subs(system, i)
-        res.append(prop.substitute(subs_i))
-    return And(res)
-
-
-def get_bmc(system, prop, k):
-    """Returns the BMC encoding at step k"""
-    init_0 = system.init.substitute(get_subs(system, 0))
-    prop_k = prop.substitute(get_subs(system, k))
-    return And(get_unrolling(system, k), init_0, Not(prop_k))
-
-def get_k_induction(system, prop, k):
-    """Returns the K-Induction encoding at step K"""
-    subs_k = get_subs(system, k)
-    prop_k = prop.substitute(subs_k)
-    return And(get_unrolling(system, k),
-               get_k_hypothesis(system, prop, k),
-               get_simple_path(system, k),
-               Not(prop_k))
-
-def check_property(system, prop):
-    """Interleaves BMC and K-Ind to verify the property."""
-    print("Checking property %s..." % prop)
-    for b in xrange(100):
-        f = get_bmc(system, prop, b)
-        print("   [BMC]    Checking bound %d..." % (b+1))
-        if is_sat(f):
-            print("--> Bug found at step %d" % (b+1))
-            return
-
-        f = get_k_induction(system, prop, b)
-        print("   [K-IND]  Checking bound %d..." % (b+1))
-        if is_unsat(f):
-            print("--> The system is safe!")
-            return
-
-
-def main():
-    # Example Transition System (SMV-like syntax)
+class PDR(object):
+    def __init__(self, system):
+        self.system = system
+        self.R = [system.init]
```

Comment:
This should be lower case. Uppercase identifiers are typically used for classes.